status-im/status-react
# Add wallet connect signing flow

fixes #19837

### Summary
 
Added the signing flow for the following RPC methods:
* `eth_sign`
* `personal_sign`
* `eth_signTypedData`
* `eth_signTypedData_v4`

The dapp data is not shown because it's yet to be merged in https://github.com/status-im/status-mobile/pull/19943


### Review notes

Have to be connected to a dapp already to be able to test the flow. The connection instructions can be found here https://github.com/status-im/status-mobile/pull/19909

### Testing notes

No manual testing needed as the functionality is hidden behind a feature flag

#### Platforms


- Android
- iOS

#### Areas that maybe impacted


##### Functional
- dapps

### Steps to test

Steps to connect via REPL can be found here: https://github.com/status-im/status-mobile/pull/19909
To test the signing:
1. Make sure the feature flag is enabled
2. Connect to dapp using REPL via [demo](https://react-app.walletconnect.com/)
3. Select any of the RPC methods (except eth_signTransaction, which is not yet implemented)
4. On the status app, a modal should appear with the details of the signing
5. Slide the auth slider at the bottom with biometric/password
6. On the demo web app, it should show a successful verification

### Recording


https://github.com/status-im/status-mobile/assets/21865759/9815d763-3707-43f9-9c2f-91c821a559b2


status: ready
